### PR TITLE
linuxPackages_testing_bcachefs: enable quota, POSIX ACL and useful debugging for tx

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

Previously, we were not enabling those two features, it does not cost us a lot to do it
and fix journald who relies on POSIX ACLs.

We also enable transaction debugging, which has a small performance cost, in exchange of a great value for the time being, see commit messages.

https://evilpiepirate.org/git/bcachefs.git/tree/fs/bcachefs/Kconfig is useful to review the options.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix [hunk 1]
```diff
@@ -32,6 +32,13 @@
         sha256 = diffHash;
       };
 
-      extraConfig = "BCACHEFS_FS m";
+      extraConfig = ''
```
Comment (lheckemann) [inline pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix:35]: Tangential, but should we move this to structuredExtraConfig? That way things like the comments here won't cause rebuilds when changed.

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix:35]: Agreed.

File: pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix [hunk 2]
```diff
@@ -32,6 +32,13 @@
...
+        BCACHEFS_FS m
+        BCACHEFS_QUOTA y
+        BCACHEFS_POSIX_ACL y
+        # Small overhead in exchange of making it easier to debug random deadlocks and hangs
+        # For now, it seems a good investment.
+        BCACHEFS_DEBUG_TRANSACTIONS y
```
Comment (misuzu) [inline pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix:41]: Isn't it [enabled](https://evilpiepirate.org/git/bcachefs.git/tree/fs/bcachefs/Kconfig) by default?

```
config BCACHEFS_DEBUG_TRANSACTIONS
	bool "bcachefs runtime info"
	depends on BCACHEFS_FS
	default y
```

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix:41]: You're definitely right… !

**Review by misuzu — APPROVED** [review]

**Pull request merged**